src/Adapter/Product/Image/ProductImagePathFactory.php

```diff
@@ -90,4 +131,26 @@ public function getHelperThumbnail(int $productId, int $shopId): string
     {
         return sprintf('%sproduct_mini_%d_%d.jpg', $this->temporaryImgDir, $productId, $shopId);
     }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return string
+     */
+    public function getImageFolder(ImageId $imageId): string
+    {
+        $path = implode('/', str_split((string) $imageId->getValue()));
+
+        return sprintf('%s%s', $this->pathToBaseDir, $path);
```

Comment:
For a simple string concatenation, I think `return $this->pathToBaseDir.$path` is 10x faster than `sprintf()` 😄  (I just did some benchmark)